components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ApiCategoriesApiServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import javax.ws.rs.core.Response;
+
+public class ApiCategoriesApiServiceImpl implements ApiCategoriesApiService {
+    private static final Log log = LogFactory.getLog(ApiCategoriesApiServiceImpl.class);
+
+    @Override
+    public Response apiCategoriesGet(MessageContext messageContext) {
+        try {
+            APIAdmin apiAdmin = new APIAdminImpl();
+            String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+            int tenantID = APIUtil.getTenantIdFromTenantDomain(tenantDomain);
+            List<APICategory> categoryList = apiAdmin.getAllAPICategoriesOfTenant(tenantID);
+            APICategoryListDTO categoryListDTO = APICategoryMappingUtil.fromCategoryListToCategoryListDTO(categoryList);
+            return Response.ok().entity(categoryListDTO).build();
+        } catch (APIManagementException e) {
+            String errorMessage = "Error while retrieving API categories";
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
+        return null;
+    }
+
+    @Override
+    public Response apiCategoriesPost(APICategoryDTO body, MessageContext messageContext) {
+        APICategory apiCategory = null;
+        try {
+            APIAdmin apiAdmin = new APIAdminImpl();
+            String userName = RestApiUtil.getLoggedInUsername();
+            apiCategory = APICategoryMappingUtil.fromCategoryDTOToCategory(body);
+            APICategoryDTO categoryDTO = APICategoryMappingUtil.
+                    fromCategoryToCategoryDTO(apiAdmin.addCategory(apiCategory, userName));
+            URI location = new URI(RestApiConstants.RESOURCE_PATH_CATEGORY + "/" + categoryDTO.getId());
+            return Response.created(location).entity(categoryDTO).build();
+        } catch (APIManagementException | URISyntaxException e) {
+            String errorMessage = "Error while adding new API Category '" + body.getName() + "' - " + e.getMessage();
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
+        return null;
+    }
+
+    @Override
+    public Response apiCategoriesApiCategoryIdPut(String apiCategoryId, APICategoryDTO body,
+                                                  MessageContext messageContext) {
+        try {
+            APIAdmin apiAdmin = new APIAdminImpl();
+            String userName = RestApiUtil.getLoggedInUsername();
+            String tenantDomain = MultitenantUtils.getTenantDomain(userName);
+            int tenantID = APIUtil.getTenantIdFromTenantDomain(tenantDomain);
+            APICategory apiCategoryToUpdate = APICategoryMappingUtil.fromCategoryDTOToCategory(body);
+            APICategory apiCategoryOriginal = apiAdmin.getAPICategoryByID(apiCategoryId);
+            if (apiCategoryOriginal == null) {
+                String errorMsg = "No api category with the given category ID exists :" + apiCategoryId;
```

Comment:
Shall we change 'api' as 'API'?

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/9702/commits/f704b669e66b63967e82ac525240beebdeed9b05